apps/test_insert_deletes_consolidate.cpp

```diff
@@ -398,23 +384,53 @@ int main(int argc, char **argv)
 
     try
     {
+        using TagT = uint32_t;
+        const bool enable_tags = true;
+        size_t dim, num_points;
+        diskann::get_bin_metadata(data_path, num_points, dim);
+
+        diskann::IndexWriteParameters params = diskann::IndexWriteParametersBuilder(L, R)
+                                                   .with_max_occlusion_size(500)
+                                                   .with_alpha(alpha)
+                                                   .with_num_threads(num_threads)
+                                                   .with_num_frozen_points(num_start_pts)
+                                                   .build();
+
+        diskann::IndexConfig index_config = diskann::IndexConfigBuilder()
+                                                .with_metric(diskann::L2)
+                                                .with_dimension(dim)
+                                                .with_max_points(max_points_to_insert)
+                                                .is_dynamic_index(true)
+                                                .with_index_write_params(params)
+                                                .with_data_type(data_type)
+                                                .with_tag_type("uint32")
```

Comment:
What is the benefit of doing that compared to the approach suggested above (which does it all at compile time and any errors are caught at compile time rather than runtime)?

Reply:
That is correct. I was thinking may be we want this to be a util function that can give us string back corresponding to its type. May be we add more types in future ??

Reply:
We can certainly do something like 
`template <typename T> const char *diskann_type_to_name() = delete;
template <> const char *diskann_type_to_name<uint32_t>()
{
    return "uint32";
}
template <> const char *diskann_type_to_name<float>()
{
    return "float";
}` for each data type. But instead I thought doing it this way avoids replication.

Reply:
I am already using this to deduce data_type and LabelT. As the index_factory logic is moved out of main.  I am deducing the type from template args.